conans/client/cmake.py

```diff
@@ -346,11 +346,11 @@ def _build_new(self, args=None, build_dir=None, target=None):
         command = "cmake --build %s" % arg_list
         self._conanfile.run(command)
 
-    def install(self):
+    def install(self, args=None, build_dir=None):
         if not self.definitions.get("CMAKE_INSTALL_PREFIX"):
             raise ConanException("CMAKE_INSTALL_PREFIX not defined for 'cmake.install()'\n"
                                  "Make sure 'package_folder' is defined")
-        self._conanfile.run("cmake --build . --target install")
+        self._build_new(args=args, build_dir=build_dir, target="install")
```

Comment:
Maybe build_dir could be used the one from configure. Which in turn can come from different source:
```
self.build_dir = build_dir or self.build_dir or self._conanfile.build_folder
```